### PR TITLE
feature: re-implement using multi-agent framework

### PR DESCRIPTION
**Pull request by marcellodebernardi**
This PR reimplements the `smolmodels` model building logic to use a multi-agent approach using the `smolagents` library. The "hard-coded" model generation algorithm has been entirely replaced by the following approach:

1. Manager agent that plans the work for building the model(s)
2. Specialised agents that plan how the ML problem should be solved, write/execute the training code, package the model, etc.

This PR significantly improves the model generation's robustness to edge cases and unforeseen inputs, because the agentic reasoning approach is more dynamic and self-correcting. For the first time, the _entire_ integration test suite passes, and anecdotally I have seen from the agent reasoning logs that the agent is capable of recovering from issues that would have killed the model build in the previous implementation.

### CODE REVIEW

File: smolmodels/internal/models/tools/metrics.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from smolagents import tool
+
+from smolmodels.internal.common.provider import Provider
+from smolmodels.internal.models.generation.planning import SolutionPlanGenerator
+
+logger = logging.getLogger(__name__)
+
+
+@tool
+def select_target_metric(task: str, provider: str) -> Dict:
```
[the remark this answers is not in the document]
Comment (marcellodebernardi) [reply smolmodels/internal/models/tools/metrics.py:17]: Doesn't play well with smolagents.

**Comment by marcellodebernardi** [comment]
Demo video for the README

https://github.com/user-attachments/assets/05ac238b-464c-457c-a63a-819bfe9a4fed

**Review by vaibs-d — APPROVED** [review]
LGTM! Thanks for deleting all of our work 🔥
